### PR TITLE
Package up library using Hatch

### PR DESCRIPTION
**Pull request by dan-katz**

### CODE REVIEW

File: upstream_delineator/py/util.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 assert CATCHMENT_PATH
 RIVER_PATH = os.getenv("RIVER_PATH")
 assert RIVER_PATH
+MEGABASINS_PATH = os.getenv("MEGABASINS_PATH")
```
Comment (tngzng) [inline upstream_delineator/py/util.py:26]: what should ppl set this too when running the delineator? 

lmk and i can update this pin messaged https://upstreamtech.slack.com/archives/C07C8RR4610/p1721681020380309?thread_ts=1721680987.163239&cid=C07C8RR4610

File: upstream_delineator/py/util.py [hunk 2]
```diff
@@ -510,18 +512,19 @@ def save_network(G: networkx.Graph, prefix: str, file_ext: str):
...
+        with open(filename, "w") as f:
+            json.dump(data, f)
+    elif file_ext == 'gml':
+        # (3) GML (Graph Modeling Language), a common graph file format.
+        networkx.write_gml(G, filename)
+    elif file_ext == 'xml':
+        # (4) GraphML is an XML-based file format for graphs.
+        networkx.write_graphml(G, filename)
+    else:
+        raise ValueError(f'Unhandled file extension {file_ext}')
```
Comment (tngzng) [inline upstream_delineator/py/util.py:530]: for a follow up - we'll never hit this bc of this bit of code above: 
```
    allowed_formats = ['pkl', 'gml', 'xml', 'json']
    if file_ext not in allowed_formats:
        print(f"River network graph not saved. Please choose one of the following formats: "
              f"{', '.join(allowed_formats)}")
        raise Warning("Did not save graph data.")
```

**Review by tngzng — APPROVED** [review]
looks good! thanks for packaging this up 🥡

File: upstream_delineator/py/util.py (hunk 1 above)
Comment (dan-katz) [reply upstream_delineator/py/util.py:26]: https://public-hydrology-data.upstream.tech/merit_hydro_vect_level2.shp

File: upstream_delineator/py/util.py (hunk 2 above)
Comment (dan-katz) [reply upstream_delineator/py/util.py:530]: I like having this here to handle the case where we add a new format to the `allowed_formats` but then forget to handle the new format in the `if elif else` conditional

Comment (tngzng) [reply upstream_delineator/py/util.py:530]: agreed! i was thinking nixing the allowed formats stuff above and centralizing it down here in the else block

**Pull request merged**